tutorials/k8s_workload.md

```diff
@@ -0,0 +1,352 @@
+
+## Set up OSC to Protect a Workload
+
+### 1. Upload Plugin
+
+Within OSC, navigate to **Manage** > **Plugins** using the left-hand menu. 
+* Upload the `SDN CONTROLLER NSC PLUGIN`.  
+* Upload the `SAMPLE MANAGER PLUGIN`.  
+
+![Upload Plugins](images/add_plugins.jpg)  
+*Upload SDN Controller and Manager Plugins*
+
+### 2. Define Virtualization Connector  
+
+Using the OSC api-doc `https://OSC_IP:8090/api-doc -> Operations for Virtualization Connectors`:
+* `POST /api/server/v1/virtualizationConnectors` with json payload:
+```json
+{
+  "name": "k8s-vc",
+  "type": "KUBERNETES",
+  "controllerIP": "string",  # IP Address of the SDN controller. Not needed if using the NSC SDN Plugin.
+  "controllerUser": "string", # User name of the SDN controller. Not needed if using the NSC SDN Plugin.
+  "controllerPassword": "string", # Password of the user of the SDN controller. Not needed if using the NSC SDN Plugin.
+  "providerIP": "string", # IP Address for the kube API service.
+  "providerUser": "string", # User name for accessing the kube API service.
+  "providerPassword": "string", # Password for the user name accessing the kube API service.
+  "softwareVersion": "v1.7",
+  "controllerType": "NSC" # Or 'Nuage' if using Nuage SDN
+}
+```
+
+You should get a response body containing an ID, similar to this `"id":VIRTUALIZATION_CONNECTOR_ID`, copy this identifier to be used on a later steps.
+
+### 3. Define Manager Connector  
+
+Using the left-hand menu, navigate to **Setup** > **Manager Connectors**, and then select **Add**.
+* Enter a name.
+* For the type, select **ISM** as described by the `SAMPLE APPLIANCE IMAGE` and the `SAMPLE MANAGER PLUGIN`.
+* Enter the IP address `1.1.1.1`, and then the credentials of `abc / 123`.
+ 	>Note: When using a real security manager, use the real IP address and credentials.
+
+![Add Manager Connector](images/add_mc.jpg)  
+*Add Manager Connector*  
+
+After adding the manager connector, ensure the **Last Job Status** is **PASSED** and that policies defined in the security manager are populated under **Polices** on the bottom-half of the page.
+
+### 4. Define Service Function  
+
+#### 4.1. Create the Appliance Model  
+Using the OSC api-doc `https://OSC_IP:8090/api-doc -> Operations for Security Functions Catalog`:  
+* `POST /api/server/v1/catalog` with json payload:
+```json
+{
+"model":"test-model",
+"managerVersion":"1.0",
+"managerType": "ISM"
+}
+```
+
+You should get a response body containing an ID, similar to this `"id":APPLIANCE_MODEL_ID`, copy this identifier to be used on later steps.
+
+
+#### 4.2. Create the Appliance Version
+Using the OSC api-doc `https://OSC_IP:8090/api-doc -> Operations for Security Functions Catalog`:  
+* `POST /api/server/v1/catalog/{applianceId}/versions` with json payload:  
```

Comment:
mention to replace Appliance_Model_ID in the payload with the id obtained in previous step